{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyPn1KVAA5OjT2QEsN2vExds"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"sDpMTLReqUMD","executionInfo":{"status":"ok","timestamp":1735530555279,"user_tz":-540,"elapsed":319203,"user":{"displayName":"유미라","userId":"17346929728541923724"}},"outputId":"0a926ee9-9e44-42aa-91c2-2fde1c711af4"},"outputs":[{"output_type":"stream","name":"stdout","text":["Warning: Looks like you're using an outdated `kagglehub` version (installed: 0.3.5), please consider upgrading to the latest version (0.3.6).\n","Downloading from https://www.kaggle.com/api/v1/datasets/download/dhiadah/car-data-train?dataset_version_number=1...\n"]},{"output_type":"stream","name":"stderr","text":["100%|██████████| 12.7G/12.7G [02:25<00:00, 93.9MB/s]"]},{"output_type":"stream","name":"stdout","text":["Extracting files...\n"]},{"output_type":"stream","name":"stderr","text":["\n"]},{"output_type":"stream","name":"stdout","text":["Path to dataset files: /root/.cache/kagglehub/datasets/dhiadah/car-data-train/versions/1\n"]}],"source":["# kaggle car dataset\n","import kagglehub\n","\n","# Download latest version\n","path = kagglehub.dataset_download(\"dhiadah/car-data-train\")\n","\n","print(\"Path to dataset files:\", path)"]},{"cell_type":"code","source":["# AI hub 자동차 차종/연식/번호판 인식용 영상\n","# import gdown, zipfile, os\n","\n","# file_id = '1cphOiHucE759f_xorWulMESer1ONjW_l'\n","\n","# gdown.download(f'https://drive.google.com/uc?id={file_id}', 'file.zip', quiet=False) # (다운로드받을것, 다운로드파일이름지정, 로그확인)\n","\n","# dir = 'cat-and-dog'\n","# os.makedirs(dir, exist_ok=True) # exist_ok -> 없으면 만들고 있어도 에러 발생 없이 넘어감\n","\n","# with zipfile.ZipFile('file.zip', 'r') as z:\n","#   z.extractall(dir)\n","\n","import gdown, zipfile, os\n","\n","file_id = '170eZ_8frqfJg2rTS0CdNCSMt7-GskdsC'\n","\n","gdown.download(f'https://drive.google.com/uc?id={file_id}', 'file.zip', quiet=False) # (다운로드받을것, 다운로드파일이름지정, 로그확인)\n","\n","dir = 'car'\n","os.makedirs(dir, exist_ok=True) # exist_ok -> 없으면 만들고 있어도 에러 발생 없이 넘어감\n","\n","\n","def extract_file(file_path, extract_to):\n","    \"\"\"압축 파일 해제 함수\"\"\"\n","    if file_path.endswith('.zip'):\n","        with zipfile.ZipFile(file_path, 'r') as zip_ref:\n","            zip_ref.extractall(extract_to)\n","            print(f\"Extracted: {file_path} to {extract_to}\")\n","    else:\n","        print(f\"Skipped (unsupported format): {file_path}\")\n","\n","def extract_all_in_folder(folder_path):\n","    \"\"\"폴더 내 모든 압축 파일 해제\"\"\"\n","    for root, _, files in os.walk(folder_path):\n","        for file in files:\n","            file_path = os.path.join(root, file)\n","            extract_to = root  # 압축 해제 폴더는 현재 파일이 위치한 폴더\n","            extract_file(file_path, extract_to)\n","\n","\n","os.makedirs(dir, exist_ok=True)\n","\n","# 3. 폴더 내 모든 압축 파일 해제\n","extract_all_in_folder(dir)\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":716},"id":"aSlXUKOI6078","executionInfo":{"status":"error","timestamp":1735540420335,"user_tz":-540,"elapsed":3529,"user":{"displayName":"유미라","userId":"17346929728541923724"}},"outputId":"dd2c4a7d-631a-4039-f340-470fa0ce227e"},"execution_count":2,"outputs":[{"output_type":"stream","name":"stderr","text":["/usr/local/lib/python3.10/dist-packages/gdown/download.py:33: MarkupResemblesLocatorWarning: The input looks more like a filename than markup. You may want to open this file and pass the filehandle into Beautiful Soup.\n","  soup = bs4.BeautifulSoup(line, features=\"html.parser\")\n"]},{"output_type":"error","ename":"FileURLRetrievalError","evalue":"Failed to retrieve file url:\n\n\tCannot retrieve the public link of the file. You may need to change\n\tthe permission to 'Anyone with the link', or have had many accesses.\n\tCheck FAQ in https://github.com/wkentaro/gdown?tab=readme-ov-file#faq.\n\nYou may still be able to access the file from the browser:\n\n\thttps://drive.google.com/uc?id=170eZ_8frqfJg2rTS0CdNCSMt7-GskdsC\n\nbut Gdown can't. Please check connections and permissions.","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mFileURLRetrievalError\u001b[0m                     Traceback (most recent call last)","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/gdown/download.py\u001b[0m in \u001b[0;36mdownload\u001b[0;34m(url, output, quiet, proxy, speed, use_cookies, verify, id, fuzzy, resume, format, user_agent, log_messages)\u001b[0m\n\u001b[1;32m    266\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 267\u001b[0;31m             \u001b[0murl\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_url_from_gdrive_confirmation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mres\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    268\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mFileURLRetrievalError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/gdown/download.py\u001b[0m in \u001b[0;36mget_url_from_gdrive_confirmation\u001b[0;34m(contents)\u001b[0m\n\u001b[1;32m     54\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m         raise FileURLRetrievalError(\n\u001b[0m\u001b[1;32m     56\u001b[0m             \u001b[0;34m\"Cannot retrieve the public link of the file. \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mFileURLRetrievalError\u001b[0m: Cannot retrieve the public link of the file. You may need to change the permission to 'Anyone with the link', or have had many accesses. Check FAQ in https://github.com/wkentaro/gdown?tab=readme-ov-file#faq.","\nDuring handling of the above exception, another exception occurred:\n","\u001b[0;31mFileURLRetrievalError\u001b[0m                     Traceback (most recent call last)","\u001b[0;32m<ipython-input-2-16f22f1a8f8f>\u001b[0m in \u001b[0;36m<cell line: 19>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0mfile_id\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'170eZ_8frqfJg2rTS0CdNCSMt7-GskdsC'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 19\u001b[0;31m \u001b[0mgdown\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdownload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf'https://drive.google.com/uc?id={file_id}'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'file.zip'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mquiet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# (다운로드받을것, 다운로드파일이름지정, 로그확인)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     20\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[0mdir\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'car'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/gdown/download.py\u001b[0m in \u001b[0;36mdownload\u001b[0;34m(url, output, quiet, proxy, speed, use_cookies, verify, id, fuzzy, resume, format, user_agent, log_messages)\u001b[0m\n\u001b[1;32m    276\u001b[0m                 \u001b[0murl_origin\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    277\u001b[0m             )\n\u001b[0;32m--> 278\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mFileURLRetrievalError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmessage\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    279\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    280\u001b[0m     \u001b[0mfilename_from_url\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mFileURLRetrievalError\u001b[0m: Failed to retrieve file url:\n\n\tCannot retrieve the public link of the file. You may need to change\n\tthe permission to 'Anyone with the link', or have had many accesses.\n\tCheck FAQ in https://github.com/wkentaro/gdown?tab=readme-ov-file#faq.\n\nYou may still be able to access the file from the browser:\n\n\thttps://drive.google.com/uc?id=170eZ_8frqfJg2rTS0CdNCSMt7-GskdsC\n\nbut Gdown can't. Please check connections and permissions."]}]}]}